stetho/src/main/java/com/facebook/stetho/inspector/elements/android/window/WindowFetcher.java

```diff
@@ -0,0 +1,103 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+
+/*
+ * 4.0.3_r1    WindowManagerImpl   private View[] mViews;
+ * 4.0.4       WindowManagerImpl   private View[] mViews;
+ *
+ * 4.1.1       WindowManagerImpl   private View[] mViews;
+ * 4.1.2       WindowManagerImpl   private View[] mViews;
+ *
+ * 4.2_r1      WindowManagerGlobal  private View[] mViews
+ * 4.2.2 r1    WindowManagerGlobal  private View[] mViews
+ *
+ * 4.3_r2.1     WindowManagerGlobal  private View[] mViews;
+ *
+ * 4.4_r1       WindowManagerGlobal   private final ArrayList<View> mViews
+ * 4.4.2_r1     WindowManagerGlobal   private final ArrayList<View> mViews
+ *
+ * 5.0.0_r2     WindowManagerGlobal   private final ArrayList<View> mViews
+ *
+ * 6.0.0_r1     WindowManagerGlobal   private final ArrayList<View> mViews
+ *
+ * 7.0.0_r1     WindowManagerGlobal   private final ArrayList<View> mViews
+ *
+ * 8.0.0_r4    WindowManagerGlobal   private final ArrayList<View> mViews
+ */
+
+public class WindowFetcher {
+	public static List<View> getWindowViews(Context context) {
+		Util.throwIfNull(context);
+		if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.JELLY_BEAN) {
+			WindowManager windowManager = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
+			Object wm = getOuter(windowManager);
+			if (wm == null) {
+				return null;
+			}
+			return getWindowViews(wm);
+		} else if (Build.VERSION.SDK_INT == Build.VERSION_CODES.JELLY_BEAN_MR1
+			|| Build.VERSION.SDK_INT == Build.VERSION_CODES.JELLY_BEAN_MR2) {
+			WindowManagerGlobal managerGlobal = WindowManagerGlobal.getInstance();
+			return getWindowViews(managerGlobal);
+
+		} else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+			return sWindow;
+		}
+		return null;
+	}
+
+	private static List<View> sWindow;
+
+	public static void hook() {
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+			WindowManagerGlobal managerGlobal = WindowManagerGlobal.getInstance();
+			try {
+				Field mViewsFiled = WindowManagerGlobal.class.getDeclaredField("mViews");
+				mViewsFiled.setAccessible(true);
+				sWindow = Collections.unmodifiableList((List<View>) mViewsFiled.get(managerGlobal));
+				mViewsFiled.setAccessible(false);
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+	}
+
+	private static Object getOuter(Object innerWM) {
+		try {
+			Field parentFiled = innerWM.getClass().getDeclaredField("mWindowManager");
+			parentFiled.setAccessible(true);
+			Object outerWM = parentFiled.get(innerWM);
+			parentFiled.setAccessible(false);
+			return outerWM;
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+		return null;
+	}
+
+	private static List<View> getWindowViews(final Object obj) {
+		try {
+			Class clz = obj.getClass();
+			final Field field = clz.getDeclaredField("mViews");
+			field.setAccessible(true);
+			return Collections.unmodifiableList(Arrays.asList((View[]) field.get(obj)));
+		} catch (Exception e) {
+			e.printStackTrace();
```

Comment:
This is a pretty nasty anti-pattern in Java and I strongly encourage you not to use it here or in other Java projects.  Instead, consider each kind of exception you actually expect, what it means, and what the code is expected to do.  For instance, if you expect that this means that we're just simply on the wrong version of the platform, perhaps we should be testing for that separately.  If you suspect that this means the manufacturer has really screwed something up by deviating from aosp standard, bark appropriately with a meaningful log message to that effect.  Each kind of exception you might get should be handled differently tho -- some of them probably should just throw RuntimeException even...